src/main/service/ResourceCleanupService.ts

```diff
@@ -0,0 +1,59 @@
+import { AWSError, S3 } from 'aws-sdk';
+import {
+  DeleteObjectOutput,
+  DeleteObjectRequest,
+  DeleteObjectsOutput,
+  DeleteObjectsRequest,
+  ObjectIdentifierList,
+} from 'aws-sdk/clients/s3';
+import { PromiseResult } from 'aws-sdk/lib/request';
+
+export const S3_BUCKET = 'promopal';
+
+/**
+ * Generic class used to capture external resource cleanup such as deleting files from S3
+ * */
+export class ResourceCleanupService {
+  private s3: S3;
+
+  constructor(s3: S3) {
+    this.s3 = s3;
+  }
+
+  /**
+   * Cleanup any external resources associated with a promotion
+   * * E.g. Delete S3 images associated with the promotion
+   * @param promotionId the id of the promotion to clean up resources for
+   * */
+  cleanupResourceForPromotion(
+    promotionId: string
+  ): Promise<PromiseResult<DeleteObjectOutput, AWSError>> {
+    const deleteObjectRequest: DeleteObjectRequest = {
+      Bucket: S3_BUCKET,
+      Key: promotionId,
+    };
+
+    return this.s3.deleteObject(deleteObjectRequest).promise();
```

Comment:
Ohh I see, using folders within buckets, this is a great idea for the future once we start supporting more and more images. Right now we only have one image per promotion. I think we should definitely do this once we start expanding, do you want to create a ticket for this?

> Also just out of curiosity- does this completely wipe everything with the key, regardless of file type?

Yeah the key must be unique, so it will only delete the single object with the requested key, regardless of file type

Reply:
So I looked into this some more, S3 does have a concept of folders, it automatically makes them folders if they share the same relative path, so basically the keys need to be paths themselves
https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-folders.html

Based on my quick research, there isn't an easy way to delete all objects in a folder, but I did see some workarounds where you can basically find all keys that start with the root folder. So for example if we grouped promotion images by userId and had the following keys
```
user1/promotion1ID
user1/promotion2ID
user1/promotion3ID
```
we would find all objects prefixed with user1, and delete those objects. In other words, we've deleted all resources associated with user1.

Reply:
Ooo yeah!! That makes sense, I can make a ticket and include the research you've done here :) I've ran into a similliar thing during work, and we use variables in our config file to maintain the keys that form the path too!